src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java

```diff
@@ -275,19 +290,24 @@ else if (msg.what == KILL_WORKER) {
 
                     ///////////////////////////
 
-                    if (queueDepth >= mConfig.getBulkUploadLimit()) {
+                    if ((returnCode >= mConfig.getBulkUploadLimit() || returnCode == MPDbAdapter.DBOutOfMemoryError) &&
+                            SystemClock.elapsedRealtime() >= mRetryAfter) {
```

Comment:
I'd like to see a test for the back-off behavior- provide a mock poster to AnalyticsMessages (that throws a ServiceUnavailableException) and somehow see that the AnalyticsMessages actually back off. I'd also like to be sure it behaves like we expect in the presence of error codes from a mock DB.